[BUG] Build error with GCC 13.1.1 on Fedora 38

**Describe the bug**

Building KeyDB on the `main ` (commit `c17b9f47acc6654bb7b4fd190b73b1f4d5188ec5`) fails with GCC 13.1.1 on Fedora 38.

**Expected behavior**

KeyDB builds successfully.

**Additional information**

Error log:

> t_zset.cpp:2039:11: ошибка: declaration of «zsetopsrc::_uT::_iterzset zsetopsrc::_uT::zset» changes meaning of «zset» [-Wchanges-meaning]
>  2039 |         } zset;
>       |           ^~~~
> t_zset.cpp:2036:17: замечание: used here to mean «typedef struct zset zset»
>  2036 |                 zset *zs;
>       |                 ^~~~
> In file included from t_zset.cpp:59:
> server.h:1794:3: замечание: объявлено здесь
>  1794 | } zset;
>       |   ^~~~
> cc1plus: замечание: unrecognized command-line option «-Wno-atomic-alignment» may have been intended to silence earlier diagnostics
> make[1]: *** [Makefile:490: t_zset.o] Error 1


Settings `CXXFLAGS` to `-Wno-changes-meaning` helps to fix the problem but I think the code (or the build scripts) should be fixed.

By the way, on the same setup but with Clang 16 I get the following error:

> anet.o: file not recognized: ae.o: file not recognized: формат файла не распознан
> формат файла не распознан
> clang-16: clang-16: error: error: linker command failed with exit code 1 (use -v to see invocation)linker command failed with exit code 1 (use -v to see invocation)
> 
> make[1]: *** [Makefile:472: keydb-diagnostic-tool] Error 1
> make[1]: *** Ожидание завершения заданий…
> make[1]: *** [Makefile:464: keydb-cli] Error 1
> ae.o: file not recognized: формат файла не распознан
> clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
> make[1]: *** [Makefile:468: keydb-benchmark] Error 1
> In file included from db.cpp:41:
> In file included from /usr/lib64/clang/16/include/xmmintrin.h:31:
> /usr/lib64/clang/16/include/mm_malloc.h:25:16: error: 'posix_memalign' is missing exception specification 'throw()'
> extern "C" int posix_memalign(void **__memptr, size_t __alignment, size_t __size);
>                ^
> ../deps/jemalloc/include/jemalloc/jemalloc.h:236:38: note: previous declaration is here
> JEMALLOC_EXPORT int JEMALLOC_NOTHROW    je_posix_memalign(void **memptr,
>                                         ^
> ../deps/jemalloc/include/jemalloc/jemalloc.h:77:29: note: expanded from macro 'je_posix_memalign'
> #  define je_posix_memalign posix_memalign
>                             ^
> 1 error generated.
> make[1]: *** [Makefile:490: db.o] Error 1
> make[1]: выход из каталога «/home/zamazan4ik/open_source/KeyDB/src»
> make: *** [Makefile:6: all] Error 2

--- a/src/t_zset.cpp
+++ b/src/t_zset.cpp
@@ -2036,7 +2036,7 @@ struct zsetopsrc {
                 zset *zs;
                 zskiplistNode *node;
             } sl;
-        } zset;
+        } iterzset;
     } iter;
 };
 
@@ -2085,7 +2085,7 @@ void zuiInitIterator(zsetopsrc *op) {
         /* Sorted sets are traversed in reverse order to optimize for
          * the insertion of the elements in a new list as in
          * ZDIFF/ZINTER/ZUNION */
-        iterzset *it = (iterzset*)&op->iter.zset;
+        iterzset *it = (iterzset*)&op->iter.iterzset;
         if (op->encoding == OBJ_ENCODING_ZIPLIST) {
             it->zl.zl = (unsigned char*)op->subject->m_ptr;
             it->zl.eptr = ziplistIndex(it->zl.zl,-2);
@@ -2118,7 +2118,7 @@ void zuiClearIterator(zsetopsrc *op) {
             serverPanic("Unknown set encoding");
         }
     } else if (op->type == OBJ_ZSET) {
-        iterzset *it = &op->iter.zset;
+        iterzset *it = &op->iter.iterzset;
         if (op->encoding == OBJ_ENCODING_ZIPLIST) {
             UNUSED(it); /* skip */
         } else if (op->encoding == OBJ_ENCODING_SKIPLIST) {
@@ -2194,7 +2194,7 @@ int zuiNext(zsetopsrc *op, zsetopval *val) {
             serverPanic("Unknown set encoding");
         }
     } else if (op->type == OBJ_ZSET) {
-        iterzset *it = &op->iter.zset;
+        iterzset *it = &op->iter.iterzset;
         if (op->encoding == OBJ_ENCODING_ZIPLIST) {
             /* No need to check both, but better be explicit. */
             if (it->zl.eptr == NULL || it->zl.sptr == NULL)

